src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+            return false;
+    }
+    return true;
+}
+
+/**
+ * @brief Checks if the provided ip address of the listen directive is valid and reads it if that is the case
+ *
+ * Because the listen directive can contain only an ip address, only a port or can contain both, it must be validated if a colon is present.
+ * If no colon is present, the port or the ip address must be valid.
+ *
+ * The function makes sure that the ip address is valid in the following ways:
+ * 1. The ip address must not contain any character other than '0'-'9' or '.'
+ * 2. The number of dots must be 3
+ * 3. The numbers within the ip address can not be smaller than 0 or greater than 255
+ *
+ * @param ip The ip address to check
+ * @return true If the ip address is valid
+ * @return false If the ip address is not valid
+ */
+bool ConfigFileParser::isIpAddressValid(const std::string& ip) const
+{
+    if (ip.find_first_not_of("0123456789.") != std::string::npos)
+        return false;
+
+    size_t firstDotIndex = ip.find('.');
+    if (firstDotIndex == std::string::npos)
+        return false;
+    size_t secondDotIndex = ip.find('.', firstDotIndex + 1);
+    if (secondDotIndex == std::string::npos)
+        return false;
+    size_t thirdDotIndex = ip.find('.', secondDotIndex + 1);
+    if (thirdDotIndex == std::string::npos)
+        return false;
+
+    std::string firstOctetStr = ip.substr(0, firstDotIndex);
+    std::string secondOctetStr = ip.substr(firstDotIndex + 1, secondDotIndex - firstDotIndex - 1);
+    std::string thirdOctetStr = ip.substr(secondDotIndex + 1, thirdDotIndex - secondDotIndex - 1);
+    std::string fourthOctetStr = ip.substr(thirdDotIndex + 1, ip.length() - thirdDotIndex - 1);
+
+    const int base = 10;
+    long firstOctet = std::strtol(firstOctetStr.c_str(), NULL, base);
+    long secondOctet = std::strtol(secondOctetStr.c_str(), NULL, base);
+    long thirdOctet = std::strtol(thirdOctetStr.c_str(), NULL, base);
+    long fourthOctet = std::strtol(fourthOctetStr.c_str(), NULL, base);
+
+    const short maxIpValue = 255;
+    const short minIpValue = 0;
+    if (firstOctet > maxIpValue || secondOctet > maxIpValue || thirdOctet > maxIpValue || fourthOctet > maxIpValue)
+        return false;
+    if (firstOctet < minIpValue || secondOctet < minIpValue || thirdOctet < minIpValue || fourthOctet < minIpValue)
+        return false;
+
+    return true;
+}
+
+/**
+ * @brief Checks if the ip address is valid and reads it if that is the case
+ *
+ * For the case that no semicolon is can be found in the listen directive, the function checks
+ * if the defined values is valid port. If that is true, the function returns.
+ * If that is not true, the function checks if it is a valid ip address and reads it.
+ *
+ */
+void ConfigFileParser::readIpAddress(const std::string& value)
+{
+    size_t colonIndex = value.find(':');
+    std::string ip;
```

Comment:
Deleted function because I have implemented a readSocket function for reading ip address and port. 
In this function I also use now ipAddress as var name instead of ip.